src/builder/modules/Generator_Builder.res

```diff
@@ -439,6 +439,67 @@ let fillBackgroundColor = (model: Model.t, fillStyle: string) => {
   }
 }
 
+let getOffset = ((x1, y1): position, (x2, y2): position) => {
+  let x1 = Belt.Int.toFloat(x1)
+  let y1 = Belt.Int.toFloat(y1)
+  let x2 = Belt.Int.toFloat(x2)
+  let y2 = Belt.Int.toFloat(y2)
+
+  let w = x2 -. x1
+  let h = y2 -. y1
+
+  /* When a line is drawn and its start and end coords are integer values, the
+  resulting line is drawn in between to rows of pixels, resulting in a line
+  that is two pixels wide and half transparent. To fix this, the line's start
+  and end positions need to be offset 0.5 pixels in the direction normal to the
+  line. The following code gets the angle of the line, and gets the components
+  for a translation in the direction perpendicular to the angle using vector
+  resolution: https://physics.info/vector-components/summary.shtml This results
+  in a fully opaque line with the correct width if the line is vertical or
+  horizontal, but antialiasing may still affect lines at other angles.
+ */
+  let angle = w === 0.0 ? Js.Math._PI /. 2.0 : Js.Math.atan2(~y=h, ~x=w, ())
+  let ox = Js.Math.sin(angle) *. 0.5
+  let oy = Js.Math.cos(angle) *. 0.5
+
+  (ox, oy)
+}
+
+let drawLine = (
+  model: Model.t,
+  (x1, y1): position,
+  (x2, y2): position,
+  ~color: string,
+  ~width: float,
+  ~pattern: array<int>,
+  ~offset: int,
+) => {
+  let (ox, oy) = getOffset((x1, y1), (x2, y2))
+
+  switch model.currentPage {
+  | None => model
+  | Some(currentPage) => {
+      let currentPage = findPage(model, currentPage.id)
```

Comment:
I think you should be able to use the `currentPage` above, no need to call `findPage()` in this case (there was a case like this in some other code, but I don't think it's needed here)